src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -149,14 +150,40 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
         workflowId: workflowId,
       });
       let status = ExecutionStatus.Unknown;
-
+      let latestDagId;
+      let latestDagResultId;
       if (!dagResultsLoading && !dagResultsError && dagResults.length > 0) {
         const latestDagResult = getLatestDagResult(dagResults);
         if (latestDagResult) {
           status = latestDagResult.exec_state.status;
+          latestDagId = latestDagResult.dag_id;
+          latestDagResultId = latestDagResult.id;
         }
       }
 
+      const nodes = useWorkflowNodes(user.apiKey, workflowId, latestDagId);
+      const nodesResults = useWorkflowNodesResults(
+        user.apiKey,
+        workflowId,
+        latestDagResultId
+      );
+
+      Object.values(nodes.operators)
+        .filter((op) => op.spec.type === 'check')
+        .forEach((op) => {
+          const artifactId = op.outputs[0]; // Assuming there is only one output artifact
+          const level = op.spec.check.level;
+          const value = nodesResults.artifacts[artifactId]?.exec_state?.status;
+          if (value === ExecutionStatus.Failed) {
+            if (level === CheckLevel.Error) {
+              status = ExecutionStatus.Failed;
+            } else if (status !== ExecutionStatus.Failed) {
+              // CheckLevel.Warning & status !== Failed
+              status = ExecutionStatus.Warning;
```

Comment:
If both error and warning exists, this will assign the status to warning if the warning one get iterated later than the error one, while the actual status shown should be error. I think an easier solution is to only do this check if the workflow status is `success` and we override it to warning based on `nodes` values

Reply:
Also, strongly recommend adding a `hasWarningCheck(nodes: NodeMap): boolean` helper in `responses/nodes` over `NodesMap` object and use it here